atomics/include/desul/atomics/cuda/CUDA_asm_exchange.hpp

```diff
@@ -1,18 +1,11 @@
-#include <cuda.h>
-
 #include <limits>
-#if (CUDA_VERSION >= 12080) && not defined(DESUL_CUDA_ARCH_IS_PRE_HOPPER)
-// CUDA 12.8 introduced 128-bit compare and swap support on Hopper+ (compute
-// capability 9.x and higher)
-#define DESUL_HAVE_CUDA_128BIT_CAS
-#endif
 
 namespace desul {
 namespace Impl {
 
 #include <desul/atomics/cuda/cuda_cc7_asm_exchange.inc>
 
-#ifdef DESUL_HAVE_CUDA_128BIT_CAS
+#ifndef DESUL_CUDA_ARCH_IS_PRE_HOPPER&& defined DESUL_HAVE_16BYTE_COMPARE_AND_SWAP
```

Comment:
That shouldn't work.
```suggestion
#if !defined DESUL_CUDA_ARCH_IS_PRE_HOPPER && defined DESUL_HAVE_16BYTE_COMPARE_AND_SWAP
```